src/main/java/roomescape/reservation/domain/Reservation.java

```diff
@@ -0,0 +1,42 @@
+package roomescape.reservation.domain;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+public class Reservation {
+
+    private final ReservationId id;
+    private final String name;
+    private final ReservationDateTime datetime;
+
+    public Reservation(final ReservationId id, final String name, final ReservationDateTime datetime) {
+        this.id = id;
+        this.name = name;
+        this.datetime = datetime;
+    }
+
+    public static Reservation createWithId(Long id, String name, LocalDateTime dateTime) {
+        return new Reservation(ReservationId.from(id), name, ReservationDateTime.from(dateTime));
+    }
+
+    public static Reservation createWithoutId(String name, LocalDateTime dateTime) {
+        return createWithId(null, name, dateTime);
+    }
```

Comment:
```suggestion
    public static Reservation of(Long id, String name, LocalDateTime dateTime) {
        return new Reservation(ReservationId.from(id), name, ReservationDateTime.from(dateTime));
    }

    public static Reservation of(String name, LocalDateTime dateTime) {
        return new Reservation(null, name, ReservationDateTime.from(dateTime));
    }
```

createWithId, createWithoutId라는 네이밍도 나쁘지 않지만, 일반적으로 도메인 객체의 정적 팩토리 메서드는 of 또는 from 같은 간결하고 직관적인 네이밍을 더 자주 사용합니다.
특히 도메인의 생성 책임이 명확히 드러나기 때문에 일관된 네이밍으로 통일하면 읽기와 유지보수 측면에서도 더 유리할 것 같아요.

Reply:
감사합니다. 되돌아보면 괜히 복잡하게 생각했던 것 같습니다.
처음에는 of로 표현하려다가, id 없이 만들어짐을 명시하고 싶어 createWithoutId라는 이름을 사용했습니다.
지금 와서 생각해보면 굳이 그것을 명시함으로써 얻는 이점이 클까? 라는 의문이 듭니다.

오히려 도메인 객체 생성에서 널리 사용되는 of, from 같은 익숙한 네이밍을 사용하는 편이, 가독성과 유지보수 측면에서 더 나은 선택이라는 제로의 의견에 깊이 공감하게 되었습니다.
저라도 나중에 해당 코드를 읽고 수정할 때, 혹은 제가 작성하지 않은 누군가의 코드를 마주할 때, of라는 메서드 명이 장기적으로 더 직관적일 것 같다는 생각이 듭니다.

(그렇다면 반대로, **정적 팩토리 메서드의 이름에 ‘행위나 상태를 명시하는 방식’**이 장점이라는 것을 학습했는데, 필요한 경우는 언제일까? 이 부분도 개인적으로는 좀 더 고민해보고 싶어졌고, 학습해보겠습니다!)

---

한 가지 추가로 여쭙고 싶은 것이 있습니다.
of(String name, LocalDateTime dateTime) 메서드 내부에서 of(null, name, dateTime)을 호출하도록 구현하는 방식은 괜찮은 구조일까요?

중복된 생성 로직을 제거하면, 필드의 생성 방식이 변경될 때 더 유연하게 대응할 수 있다고 생각했습니다.

// 공통된 생성 로직을 직접 작성
public static Reservation of(String name, LocalDateTime dateTime) {
    return new Reservation(null, name, ReservationDateTime.from(dateTime, someString, anyInt, stringDateTime));
}

// 기존 정적 팩토리 메서드를 재사용
public static Reservation of(String name, LocalDateTime dateTime) {
    return Reservation.of(null, name, dateTime);
}

이러한 맥락에서 볼 때 중복 제거와 명시성 사이에서 어떻게 균형을 잡는 것이 좋을지 의견이 궁금합니다.

Reply:
좋은 고민입니다!
저도 내부적으로 of 호출해서 재사용 하는 방식은 중복 제거하고 변경에도 용이할 것 같아서 좋은 선택이라고 생각합니다 ✈️

Reply:
감사합니다!